invenio_curations/services/diff.py

```diff
@@ -0,0 +1,248 @@
+        "resubmit": _("Record was resubmitted for review with the following changes:"),
+        "update_while_critiqued": _("Record was updated after changes were requested!"),
+        "update_while_review": _(
+            "Record was updated! Please check the latest changes."
+        ),
+        "default": _("Action triggered comment update"),
+    }
+    _added = _("Added:")
+    _changed = _("Changed:")
+    _removed = _("Removed:")
+
+    def __init__(self, diffs=None, configured_elements=None, comment_template=None):
+        """Constructs."""
+        self._diffs = diffs
+        self._configured_elements = configured_elements
+        self._comment_template = comment_template
+
+    def _map_one_diff(self, raw_diff):
+        """Return the class of a specific diff.
+
+        Parameters
+        ----------
+        raw_diff : tuple
+            This should be a tuple resulted from a dictdiffer.diff operation
+
+        Returns
+        -------
+        class
+            the wrapper class of the given diff
+        """
+        for element in self._configured_elements:
+            if element().match_diff_key(raw_diff):
+                return element
+
+        return DiffElement
+
+    def map_and_build_diffs(self, raw_diffs):
+        """Maps the diffs to their specific wrapper class, instantiates them and adds them to the list.
+
+        Parameters
+        ----------
+        raw_diffs: list
+            This should be a list of tuples resulted from a dictdiffer.diff operation
+        """
+        if not isinstance(raw_diffs, list):
+            raise DiffException()
+
+        self._diffs = []
+        for raw_diff in raw_diffs:
+            self._diffs.append(self._map_one_diff(raw_diff)(raw_diff))
+
+    def validate_and_cleanup(self):
+        """Removes structures that shouldn't be in diffs list."""
+        to_remove = []
+
+        for diff in self._diffs:
+            if not diff.validate_and_cleanup():
+                to_remove.append(diff)
+
+        for remove in to_remove:
+            self._diffs.remove(remove)
+
+    def to_html(self, action):
+        """Renders the diffs using the configured template.
+
+        Parameters
+        ----------
+        action: string
+            Action is used to get the title of the comment.
+
+
+        Returns
+        -------
+        str
+            the rendered comment
+        """
+        if action not in self._known_actions:
+            action = "default"
+
+        adds = []
+        changes = []
+        removes = []
+
+        self.validate_and_cleanup()
+        for diff in self._diffs:
+            update, _, _ = diff.get_diff()
+            if update == "add":
+                adds.append(diff.to_html())
+            elif update == "change":
+                changes.append(diff.to_html())
+            elif update == "remove":
+                removes.append(diff.to_html())
+
+        try:
+            return self._comment_template.render(
```

Comment:
instead of using `template.render()`, i think you want to call `flask.render_template(template_name, ...)` here – this takes care of resolving the correct template and thus eliminates the need for some code in this PR

instead, i would suggest passing the template name via the constructor of this class